src/app/site/site.component.ts

```diff
@@ -17,13 +17,25 @@ export class SiteComponent implements OnInit {
   isMobile = false;
   curWidth: number;
   curHeight: number;
+  spinnerEnabled = false;
+  blockedDocument = false;
 
   constructor(@Inject(WINDOW) private window, private authService: AuthService,
               private responsiveService: ResponsiveService,
-              private router: Router, private pageService: PageService) {
+              private router: Router, private pageService: PageService, private spinnerService: SpinnerService) {
   }
 
   ngOnInit() {
+    this.spinnerService.isblock$.subscribe(is_block => {

```

Comment:
these two subscription could be merged as whenever u are subscribe on isBlock$, u are also subscribing on isSpinner$